Assets/HoloToolkit/UX/Scripts/Buttons/CompoundButtonIcon.cs

```diff
@@ -0,0 +1,313 @@
+//
+using System.Collections;
+using UnityEngine;
+
+namespace HoloToolkit.Unity.Buttons
+{
+    [ExecuteInEditMode]
+    [RequireComponent(typeof(CompoundButton))]
+    public class CompoundButtonIcon : ProfileButtonBase<ButtonIconProfile>
+    {
+        /// <summary>
+        /// Turns off the icon entirely
+        /// </summary>
+        public bool DisableIcon = false;
+
+        /// <summary>
+        /// Disregard the icon in the profile
+        /// </summary>
+        public bool OverrideIcon = false;
+
+        /// <summary>
+        /// Icon to use for override
+        /// </summary>
+        public Texture2D IconOverride
+        {
+            get
+            {
+                return iconOverride;
+            }
+            set
+            {
+                iconOverride = value;
+                SetIconName(string.Empty);
+            }
+        }
+
+        /// <summary>
+        /// Property to use in IconMaterial for alpha control
+        /// Useful for animating icon transparency
+        /// </summary>
+        public float Alpha
+        {
+            get
+            {
+                return alphaTarget;
+            }
+            set
+            {
+                if (alphaTarget != value)
+                {
+                    alphaTarget = value;
+                    if (Application.isPlaying)
+                    {                        
+                        if (Mathf.Abs (alpha - alphaTarget) < 0.01f)
+                            return;
+
+                        if (updatingAlpha)
+                            return;
+
+                        if (gameObject.activeSelf && gameObject.activeInHierarchy)
+                        {
+                            // Update over time
+                            updatingAlpha = true;
+                            StartCoroutine(UpdateAlpha());
+                        }
+                        else
+                        {
+                            // If we're not active, just set the alpha immediately
+                            alpha = alphaTarget;
+                            RefreshAlpha();
+                        }
+                    }
+                    else
+                    {
+                        alphaTarget = value;
+                        alpha = alphaTarget;
+                        RefreshAlpha();
+                    }
+                }
+            }
+        }
+
+        public MeshRenderer IconRenderer
+        {
+            get
+            {
+                return targetIconRenderer;
+            } set
+            {
+                targetIconRenderer = value;
+            }
+        }
+
+        public MeshFilter IconMeshFilter
+        {
+            get
+            {
+                if (IconRenderer != null)
+                {
+                    return IconRenderer.GetComponent<MeshFilter>();
+                }
+                return null;
```

Comment:
we could simplify this to be `return IconRenderer != null ? IconRenderer.GetComponent<MeshFilter>() : null;`